packager/media/base/widevine_key_source.cc

```diff
@@ -213,13 +213,17 @@ Status WidevineKeySource::GetKey(const std::vector<uint8_t>& key_id,
 }
 
 Status WidevineKeySource::GetCryptoPeriodKey(uint32_t crypto_period_index,
+                                             uint32_t crypto_period_duration_in_seconds,
                                              const std::string& stream_label,
                                              EncryptionKey* key) {
   DCHECK(key_production_thread_.HasBeenStarted());
   // TODO(kqyang): This is not elegant. Consider refactoring later.
   {
     base::AutoLock scoped_lock(lock_);
     if (!key_production_started_) {
+      // Record crypto period duration, which should not change once
+      // the key rotation has been enabled.
+      crypto_period_duration_in_seconds_ = crypto_period_duration_in_seconds;
```

Comment:
Can you return an error if |crypto_period_duration_in_seconds_| changes, i.e.

    if (!key_production_started_) {
      ...
      crypto_period_duration_in_seconds_ = crypto_period_duration_in_seconds;
      ...
    } else if (crypto_period_duration_in_seconds_ !=
               crypto_period_duration_in_seconds) {
      return Status(error::INVALID_ARGUMENT,
                    "Crypto period duration should not change.");
    }

Reply:
Yes, this is much better than just a comment